index.bs

```diff
@@ -913,10 +927,29 @@ When the {{MLContext/[[contextType]]}} is set to [=context type/default=] with t
 
 <details open algorithm>
   <summary>
-    To <dfn>validate buffer with descriptor</dfn> given {{ArrayBufferView}} |bufferView| and {{MLOperandDescriptor}} |descriptor|, run the following steps:
+    To <dfn>validate buffer with descriptor</dfn> given {{AllowSharedBufferSource}} |bufferSource| and {{MLOperandDescriptor}} |descriptor|, run the following steps:
+  </summary>
+    1. If |bufferSource|'s [=BufferSource/byte length=] is not equal to |descriptor|'s [=MLOperandDescriptor/byte length=] return false.
+    1. Switch on |bufferSource|:
```

Comment:
Maybe "Switch on the type of ..." 

(Usually "switch" is operating on a value. https://garykac.github.io/procspec/#switch would even suggest that the cases be phrases, e.g. "If buffer source is a ..." but I think that's overkill here.)